command/agent/agent.go

```diff
@@ -356,6 +357,29 @@ func (a *Agent) clientConfig() (*clientconfig.Config, error) {
 	return conf, nil
 }
 
+// setupKeyring is used to initialize and load keyring during server startup
+func (a *Agent) setupKeyring(config *nomad.Config) error {
+	file := filepath.Join(a.config.DataDir, serfWANKeyring)
```

Comment:
https://www.nomadproject.io/docs/internals/gossip.html here I found that all the Nomad servers participate in a single global WAN pool, and also `SerfConfig` uses WAN-specific defaults (https://github.com/hashicorp/nomad/blob/e0a64086764f3021068f637a6c5c4bbf51e2cce8/nomad/config.go#L259). Would you mind briefly explaining this change request?